### PR TITLE
Switch to circular flags

### PR DESCRIPTION
**Pull request by jonstieglitz**
## Context

image

The circular flags will only be used with `np-theme-personal` - otherwise you'll get the old rectangular flags

The `Files changed` is real messy because of my prettier and because I renamed `/flags` to `/rectangular-flags`. Here's a list of the changes I made:

- switch to yarn which allows us to consume wise-atoms. npm requires a package.json in the consumed repo, yarn doesn't
- update config.yml to use yarn install
- added wise-atoms as a devDependency. this is important because otherwise it breaks other projects builds. somehow being a devDependency is enough for the package to be there during the circleCI build and push to npm. 🙏 
- added script to copy over the flags from `node_modules/wise-atoms/flags` to `dist/flags` **this is a big one because it's how everyone else will get the flags without needing to access wise-atoms**
- renamed the flags folder as `rectangular-flags`
- updated the `background-image` `url` for some of the currency-flags to point to the new wise-atoms SVG. we don't have all of them so most flags are still the old versions
- for the currencies that don't have a new circular flag, I used CSS to turn them in to circles and to stretch them in so that they're centered
Screen Shot 2022-10-05 at 4 39 38 PM  

- add dist folder to gitignore and delete dist folder from github 
- update README, CHANGELOG, and bumped version number to 4.0
- changed sizes of circular flags in css:
  - default: 24px x 16px -> 24px x 24px
  - small: 16px x 10px -> 16px x 16px
  - large: 36px x 24px -> 36px x 36px
  - extra-large: 49px x 32px - > 48px x 48px

 In [neptune-web](https://github.com/transferwise/neptune-web/pull/1751/files#) I had to remove this box-shadow and add an outline
image

I tested by building locally and copying over the dist folder to neptune-web. See image above

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,22 +19,21 @@ This package outputs two CSS files (minified and unminified) with classes for cu
 <div class="currency-flag currency-flag-usd"></div>
 ```
 
-<img src="https://wise.com/public-resources/assets/flags/rectangle/usd.png">
```
Comment (richardmccartney) [inline README.md:21]: These type of currency flags will still be available correct? Should we mention that at all in the docs?

File: src/styles.css [hunk 2]
```diff
@@ -1,181 +1,668 @@
 .currency-flag {
   display: inline-block;
   width: 24px;
-  height: 16px;
+  height: 24px;
   background-size: cover;
-  background-image:url('https://wise.com/public-resources/assets/flags/rectangle_fallback.png');
+  background-image: url("./flags/wise.svg");
```
Comment (richardmccartney) [inline src/styles.css:6]: As the `background-image` points to this path, will our teams need to do any extra steps to ensure it works on their applications? 

Can we test this at all within a app just to see what edge cases might happen when we make this switch 👍

**Review by richardmccartney — COMMENTED** [review]
PR looks good just got a few questions:

- Can we test this outside of `neptune-web` just want to see how the change of file assets will affect our web apps, what steps will they need to take to update and change to the new flags
- Are we able to test this also works well with in `styleguide-components`? Just want to ensure we have good backwards compatibility on that one.
- Can we add the ability to make a `beta` build of this change? We can then look to adding it with in `neptune-web` under the `next` branch 👍

File: .circleci/config.yml [hunk 3]
```diff
@@ -37,7 +37,7 @@ jobs:
           command: npm run build
       - run:
           name: Publish on npm
-          command: npm publish --access=public
+          command: npm publish --access=public --tag=beta
```
Comment (dozortsev) [inline .circleci/config.yml:40]:
```suggestion
          command: npm publish --access=public --tag beta
```

File: package.json [hunk 4]
```diff
@@ -1,16 +1,17 @@
 {
   "name": "currency-flags",
-  "version": "3.4.0",
+  "version": "4.0.0",
```
Comment (dozortsev) [inline package.json:3]:
```suggestion
  "version": "4.0.0-beta.1",
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -63,11 +63,11 @@ workflows:
             - test
```
Comment (dozortsev) [inline .circleci/config.yml:63]:
```suggestion
            - dependencies
```

**Review by dozortsev — COMMENTED** [review]
don't forget reset these three commits before merging into `main`

- https://github.com/transferwise/currency-flags/pull/63/commits/4ac6f495e9438893b72f1efa2c58fb65775a067f
- https://github.com/transferwise/currency-flags/pull/63/commits/0bf7619b0698ab15ebd1785b433649542e8d5541
- https://github.com/transferwise/currency-flags/pull/63/commits/64430a3584b82c3e523439d1bd3fc270ede98f58

File: src/styles.css (hunk 2 above)
Comment (jonstieglitz) [reply src/styles.css:6]: The images are bundled in `currency-flags` and no wise-atoms is included so it works quite well.

I tested it in balance-flows and it looks funky because I didn't also consume the CSS change i made in neptune-web but overall works great. We need to discuss sizing - i'll do that soon:

image

**Comment by jonstieglitz** [comment]
> * Can we test this outside of `neptune-web` just want to see how the change of file assets will affect our web apps, what steps will they need to take to update and change to the new flags

See above in balance-flows. Works pretty good

> * Are we able to test this also works well with in `styleguide-components`? Just want to ensure we have good backwards compatibility on that one.

Still working on this one

> * Can we add the ability to make a `beta` build of this change? We can then look to adding it with in `neptune-web` under the `next` branch 👍

Great idea - Anton helped me set it up: https://github.com/transferwise/neptune-web/pull/1651

**Comment by jonstieglitz** [comment]
https://transferwise.github.io/styleguide-components/branch/circular-flags/index.html#select

I got it working on styleguide-compnents too! https://github.com/transferwise/styleguide-components/pull/396

I've got to run. I'll fill out the PR tomorrow.

**Review by dozortsev — APPROVED** [review]
